### ISSUE TITLE
Realm failure when attaching HDF5 external resources

### ISSUE THREAD

**Issue by bandokihiro** [StanfordLegion/legion#1014]
On the most recent `control_replication`, I have this assertion error when running on more than 1 node.
```
FATAL: class N5Realm15HDF5LayoutPieceILi1ExEE not registered with serdez helper for N5Realm19InstanceLayoutPieceILi1ExEE
solver: /home/ihme/bandok/Softwares/legion/runtime/realm/serialize.inl:745: static bool Realm::Serialization::PolymorphicSerdezHelper<T>::serialize(Realm::Serialization::ByteCountSerializer&, const T&) [with T = Realm::InstanceLayoutPiece<1, long long int>]: Assertion `0' failed.
```
These lines in the top-level task seem to be the reason
```
AttachLauncher attach_launcher(EXTERNAL_HDF5_FILE, restart_lr, restart_lr, false, false);
attach_launcher.attach_hdf5(file_name.c_str(), field_map, LEGION_FILE_READ_WRITE);
restart_pr = runtime->attach_external_resource(ctx, attach_launcher);
```
It works fine on this commit 1f518fb3622771ffd1d7df2fca8dbfac895871de

Attached are backtraces.

[gdb0.txt](https://github.com/StanfordLegion/legion/files/5939738/gdb0.txt)
[gdb1.txt](https://github.com/StanfordLegion/legion/files/5939739/gdb1.txt)

**Comment by streichler** [comment]
This is probably broken in master as well.  Will try to get it fixed shortly.

**Comment by bandokihiro** [comment]
Fixed in commit 22d935c832033f123d4883d46fe504cdbaecce9a
